File: app/src/main/java/com/qiaobateach/mvp/MVPDemoActivity.java
package com.qiaobateach.mvp;


import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.Toast;

import com.qiao_basemodule.mvp.BaseActivity;
import com.qiao_basemodule.mvp.Data;

import java.util.List;

public class MVPDemoActivity extends BaseActivity<IDemoBaseView,DemoPresent<IDemoBaseView>> implements IDemoBaseView{


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        present.fetchData();
    }

    @Override
    public void showLoading() {
        Toast.makeText(this,"loading...",Toast.LENGTH_LONG).show();
    }

    @Override
    public void cancelLoading() {
        Toast.makeText(this,"success",Toast.LENGTH_LONG).show();
    }

    @Override
    public void showData(List<Data> data) {
        Toast.makeText(this,"show UI (bind data)",Toast.LENGTH_LONG).show();
    }

    @Override
    protected DemoPresent<IDemoBaseView> createPresent() {
        return new DemoPresent<>();
    }
}

File: app/src/main/java/com/qiaobateach/mvp/MVPDemoActivity.kt
package com.qiaobateach.mvp


import android.os.Bundle
import android.widget.Toast

import com.qiao_basemodule.mvp.BaseActivity
import com.qiao_basemodule.mvp.Data

class MVPDemoActivity : BaseActivity<IDemoBaseView, DemoPresent<IDemoBaseView>>(), IDemoBaseView {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        present.fetchData()
    }

    override fun showLoading() {
        Toast.makeText(this, "loading...", Toast.LENGTH_LONG).show()
    }

    override fun cancelLoading() {
        Toast.makeText(this, "success", Toast.LENGTH_LONG).show()
    }

    override fun showData(data: List<Data>) {
        Toast.makeText(this, "show UI (bind data)", Toast.LENGTH_LONG).show()
    }

    override fun createPresent(): DemoPresent<IDemoBaseView> {
        return DemoPresent()
    }


}
